### PR TITLE
chore(ci): verify artifacts MONGOSH-1695

### PR DESCRIPTION
**Pull request by mabaasit**

### CODE REVIEW

File: .evergreen/evergreen.yml.in [hunk 1]
```diff
@@ -1734,6 +1789,11 @@ buildvariants:
   - name: draft_publish_release
     display_name: "Draft/Publish Release"
     run_on: ubuntu2004-small
+    depends_on:
+      <% for (const { packages } of RELEASE_PACKAGE_MATRIX) {
+       for (const { name: packageVariant } of packages) { %>
+      - name: verify_artifact_<% out(packageVariant.replace(/-/g, '_')) %>
+      <% } } %>
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:1796]: We generally don't set `depends_on` in the build variants section, there's an existing one for the `release_draft` task specifically that shows what the task dependencies look like:

https://github.com/mongodb-js/mongosh/blob/590e5725265723d752048cff37b29209ac4cfd15/.evergreen/evergreen.yml.in#L1267-L1279

So I think it should be enough to tag the `verify_artifact_...` tasks with the `smoke-test` tag

File: .evergreen/verify-packaged-artifact.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  # RPM packages are signed using gpg and the signature is embedded in the package.
+  # Here, we need to import the key in `rpm` and then verify the signature.
+  echo "Importing key into rpm"
+  rpm --dbpath "$GPG_HOME" --import $MONGOSH_KEY > "$TMP_FILE" 2>&1
+  # Even if the file is not signed, the command below will exit with 0 and output something like: digests OK
+  # So we need to check the output of the command to see if the file is signed successfully.
+  echo "Verifying $1 using rpm"
+  output=$(rpm --dbpath "$GPG_HOME" -K $ARTIFACTS_DIR/$1)
+
+  # Check if the output contains the string "pgp md5 OK"
```
Comment (addaleax) [inline .evergreen/verify-packaged-artifact.sh:51]:
```suggestion
  # Check if the output contains the string "digests signatures OK"
```

although admittedly the entire comment is just very close to describing the literal content of the next line anyway :)

File: .evergreen/evergreen.yml.in [hunk 3]
```diff
@@ -1365,6 +1391,36 @@ buildvariants:
       <% } } } %>
       - name: sign_artifact_win32_x64
       - name: sign_artifact_win32msi_x64
+  - name: verify_debian_artifact
+    display_name: "Ubuntu 20.04 x64 (Verification)"
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:1395]:
```suggestion
    display_name: "Ubuntu 20.04 x64 (Signature Verification)"
```

(or the suggestion below this one, I have a mild preference for that one but either way is fine by me)

File: .evergreen/evergreen.yml.in [hunk 4]
```diff
@@ -1365,6 +1391,36 @@ buildvariants:
...
+    run_on: ubuntu2004-small
+    tasks:
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:1397]:
```suggestion
```

alternative suggestion: just make this part of `linux_package`, having it grouped with the packaging and signing tasks is actually pretty reasonable imo

File: .evergreen/evergreen.yml.in [hunk 5]
```diff
@@ -1365,6 +1391,36 @@ buildvariants:
...
+      <% for (const { packages } of RELEASE_PACKAGE_MATRIX) {
+          for (const { name: packageVariant } of packages) {
+            if (packageVariant.startsWith('deb')) { %>
+      - name: verify_artifact_<% out(packageVariant.replace(/-/g, '_')) %>
+      <% } } } %>
+  - name: verify_rhel_artifact
+    display_name: "RHEL (Verification)"
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:1404]:
```suggestion
    display_name: "RHEL (Signature Verification)"
```

File: .evergreen/evergreen.yml.in [hunk 6]
```diff
@@ -1365,6 +1391,36 @@ buildvariants:
...
+    run_on: rhel80-small
+    tasks:
+      <% for (const { packages } of RELEASE_PACKAGE_MATRIX) {
+          for (const { name: packageVariant } of packages) {
+            if (packageVariant.startsWith('rpm')) { %>
+      - name: verify_artifact_<% out(packageVariant.replace(/-/g, '_')) %>
+      <% } } } %>
+  - name: verify_windows_artifact
+    display_name: "Windows (Verification)"
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:1413]:
```suggestion
    display_name: "Windows (Signature Verification)"
```

File: .evergreen/evergreen.yml.in [hunk 7]
```diff
@@ -1365,6 +1391,36 @@ buildvariants:
...
+    run_on: windows-vsCurrent-small
+    tasks:
+      - name: verify_artifact_win32_x64
+      - name: verify_artifact_win32msi_x64
+  - name: verify_mac_artifact
+    display_name: "MacOS (Verification)"
```
Comment (addaleax) [inline .evergreen/evergreen.yml.in:1419]:
```suggestion
    display_name: "MacOS (Signature Verification)"
```

**Review by addaleax — COMMENTED** [review]
Looks great!

File: .evergreen/evergreen.yml.in (hunk 4 above)
Comment (mabaasit) [reply .evergreen/evergreen.yml.in]: addressed feedback in [`eea7fde` (#1961)](https://github.com/mongodb-js/mongosh/pull/1961/commits/eea7fde4c7d583796a4edf0f2dff874073889117)

And i just noticed **just make this** in your comment.

**Review by addaleax — APPROVED** [review]

**Pull request merged**